pertpy/tools/_distances.py

```diff
@@ -0,0 +1,407 @@
+from __future__ import annotations
+
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from ot import unif
+from ot.gromov import gromov_wasserstein
+from scipy.spatial.distance import cdist
+from sklearn.metrics import pairwise_distances
+from statsmodels.stats.multitest import multipletests
+from tqdm import tqdm
+
+"""
+DEV NOTES:
+
+User interface:
+- A function to get distances between any two or more groups of cells with a
+distance metric of choice
+- get p-values for permutation test with a distance metric of choice for all or
+    some of their groups
+
+Internally:
+- There will be two types of PermutationTest classes:
+    - One that recomputes the pairwise distance matrix every time and works for
+      any metric.
+    - One that uses the precomputed pairwise distance matrix and works only for
+    metrics that can be called on the pairwise distance matrix. Will be faster.
+NOTE: This is a bit redundant, maybe merge? Make it Fast automatically
+if Metric is callable on P?
+NOTE: Maybe move the tests to a separate file? (e.g. _tests.py) Might lead to
```

Comment:
You can create a module for the distances only:

`tools/_distances/_distances.py`
and 
`tools/_distances/_distance_tests.py`